### ISSUE TITLE
Confusing documentation options for some plugins

### ISSUE THREAD

**Issue by purbon** [elastic/logstash#4013]
Hi,
  from https://github.com/logstash-plugins/logstash-filter-drop/issues/5 we noticed that because of https://github.com/elastic/logstash/blob/master/lib/logstash/filters/base.rb#L50-L125 where configuration params like `remove_field`  or `add_tag` are defined for all filters, the configuration options (auto generated) for some of them looks misleading.

In the concrete example of `logstash-filter-drop` if this drops, or not, an entire event, having remove_tag, or add_field is just misleading.

Some other filters where this general configuration might be misleading are:
- https://github.com/logstash-plugins/logstash-filter-sleep
- https://github.com/logstash-plugins/logstash-filter-ruby
- https://github.com/logstash-plugins/logstash-filter-prune
- https://github.com/logstash-plugins/logstash-filter-cipher
- https://github.com/logstash-plugins/logstash-filter-checksum

and probably some others.

The most simple way to implement this is by having a raw base filter class + one with this configs out, there is also the option to refactor/filter out this config options in the selected filters.

**Comment by sfontana** [comment]
Yes that's very confusing indeed!

**Comment by purbon** [comment]
@ph what do you think about this?

**Comment by ph** [comment]
Indead this is misleading thanks for raising the issue again, even with the rewrite from #5715 it wont fix that problem since the doc tool is dumb and will blindly follow the inheritance chain to generate the doc and will add theses option even if the plugin don't support it.

Your proposed solutions:
A) The most simple way to implement this is by having a raw base filter class + one with this configs out.

B) there is also the option to refactor/filter out this config options in the selected filters.

A. would be the more rational way of fixing this, because if a plugin doesn't have a feature it shouldn't have the option in the first place.  Using these options should make the plugin **FAILS** before starting LS. To do so, we could create a base class and a module that add theses features. The problem with this solution I expect backporting to be *painful*.

B) Even if I already find that the doc gen is a bit of hacky, It might make sense to add for now this logic there, which would make backward compatibily work.


We should certainly do A, even if we do only B. But I am not sure how to do it with backward compatibility in mind, you have any solution for that?

**Comment by purbon** [comment]
@ph really have no solution in mind, I reported this basically as an issue to be tracked with we engaged with having more robust documentation tool. But I do understand it might be out of scope from your current work at #5715.

It would be nice to engage in something like A, I guess something more for the list of debt to be cleanup, right? @suyograo what do you think?

**Comment by ph** [comment]
@purbon Well My code is flexible enought B. could be done :)

**Comment by purbon** [comment]
@ph let's do it the right way, without hurry 👍, end of the day, there is no hurry! you said A was the best way in your opinion, right? Do you think we could do that in parallel?

**Comment by ph** [comment]
@purbon I've looked at your list again and in their code.

All of theses plugins call `filter_matched` which would drop `remove_field` or `add_tag`, even the **drop** filter would do it but the event would be cancelled and wont make it to the next filter.

So it is really an issue?
